.github/workflows/tests.yml

```diff
@@ -24,15 +24,29 @@ jobs:
       fail-fast: false
       matrix:
         os: ["windows-latest", "ubuntu-latest", "macos-latest"]
-        python-version: ["3.8", "3.9", "3.10", "3.11"]
+        environment: ["3.8", "3.9", "3.10", "3.11"]
+        extra: [null]
         exclude:
-            - os: "macos-latest"
-              python-version: "3.8"
-            - os: "macos-latest"
-              python-version: "3.10"
+          - os: "macos-latest"
+            environment: "3.8"
+          - os: "macos-latest"
+            environment: "3.10"
+        include:
+          # Minimum dependencies
+          - os: "ubuntu-latest"
+            environment: "mindeps-array"
+          - os: "ubuntu-latest"
+            environment: "mindeps-dataframe"
+          - os: "ubuntu-latest"
+            environment: "mindeps-distributed"
+          - os: "ubuntu-latest"
+            environment: "mindeps-non-optional"
+          # Pyarrow strings turned on
```

Comment:
While adding this new job, I realized we can easily move our `mindeps` builds over to the main testing workflow (currently they're in a separate `additional.yml` workflow). This isn't strictly needed for adding the pyarrow job, but I thought I'd include it here anyways.